sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -355,14 +355,14 @@ public ConfigurationClientBuilder configuration(Configuration configuration) {
     }
 
     /**
-     * Sets the {@link HttpPipelinePolicy} that is used when each request is sent.
+     * Sets the {@link RetryPolicy} that is used when each request is sent.
      *
      * The default retry policy will be used if not provided {@link ConfigurationClientBuilder#buildAsyncClient()}
      * to build {@link ConfigurationAsyncClient} or {@link ConfigurationClient}.
      * @param retryPolicy user's retry policy applied to each request.
      * @return The updated ConfigurationClientBuilder object.
      */
-    public ConfigurationClientBuilder retryPolicy(HttpPipelinePolicy retryPolicy) {
+    public ConfigurationClientBuilder retryPolicy(RetryPolicy retryPolicy) {
```

Comment:
This breaks public API and won't be allowed. The old method either needs to be deprecated and this put in it's place or remmain as is.

Reply:
ok. I will revert back the change